### ISSUE TITLE
page_snrifar try/except prevents diagnosis of missing foreground

### ISSUE THREAD

**Issue by tdent** [gwastro/pycbc#2178]
There are foreground events in an hdf file I am trying to plot but the 'cumulative' plot can't find them: see eg https://www.atlas.aei.uni-hannover.de/~tdent/LSC/o2/rates/catalog/H1L1-SNRIFAR_CUMULATIVE_O2.png .
However because there is a try-except around the code that gets the foreground triggers I can't see why it doesn't find them.  I don't see why the try-except is needed anyway.  

Also (what I suspect is causing the issue) there is an unused variable 'fap_end' defined in the cumulative case foreground trigger retrieval.  Why is this 'fap_end' here at all?

**Comment by tdent** [comment]
Also, the ymax that is plotted is 10 times the specified maximum (`pylab.ylim(plot_ymin, plot_ymax * 10.0)`.  This is inappropriate if the user specifies ymax.

**Comment by ahnitz** [comment]
@tdent Yep, I agree with your last point. I think the line you  refer to is also a holdover from when the dimensions where more automatically guessed.

**Comment by tdent** [comment]
I think it is reasonable for the code to automatically put some headroom above the maximum plotted value if not otherwise specified ..

**Comment by GarethCabournDavies** [comment]
`page_snrifar` looks like it finds foreground events in 'cumulative' mode now (from looking at a recent O4 open box)

However I think that this is the try/except statement [here](https://github.com/gwastro/pycbc/blob/a2aafefb3f8c6cef64f0288b5ff144f81bd440cc/bin/plotting/pycbc_page_snrifar#L132) that was causing the problems.

I think that should be checking for `KeyError`s in lines 135-138, but wanted to check before implementing that
